tests/src/image.rs

```diff
@@ -526,3 +527,51 @@ impl ReadbackBuffers {
                 .unwrap_or(true)
     }
 }
+
+pub fn capture_rgba_u8_texture(
```

Comment:
Couldn't you use the already existing ReadbackBuffers to do this (and the SPI tests should use them too?)

Reply:
This is a bit tricky. The entire problem is the COPY_BYTES_PER_ROW_ALIGNMENT requirement. Both in the scissors test and in the spi, the tests render the 2x2 texture. At first, I thought that it will be a good idea to support copying textures that do not meet the requirement in the readback buffers implementation, but when I started implementing that, at some point it felt that readback buffers needed their own tests as the implementation was getting too complicated (multiple types of textures, multiple ways to copy them, etc..) Personally, I like to keep tests as simple as possible, so I decided the best approach to this will be just to rewrite tests to meet the alignment requirement. What do you think about this?

Reply:
Shouldn't you be able to manage the COPY_BYTES_PER_ROW_ALIGNMENT by purely byte manipulation?

When computing the buffer size and executing the copy round up the `bytes_per_row` to `COPY_BYTES_PER_ROW_ALIGNMENT`. Then using the real bytes_per_row information, copy only the chunks that have real data in them into the vector.

I would prefer to keep the tests small, as it helps with understanding what exactly is getting tested.

Reply:
"I would prefer to keep the tests small, as it helps with understanding what exactly is getting tested."
I agree with this.

"Shouldn't you be able to manage the COPY_BYTES_PER_ROW_ALIGNMENT by purely byte manipulation?"
Yes, and that was done in the original `capture_rgba_u8_texture` function that was written in the SPI test.

I think that I may have overanalyzed your initial comment. Just to make it clear, you want to keep the `capture_rgba_u8_texture` function in the image.rs, but use ReadbackBuffers in the implementation, am I right?

Reply:
Ideally ReadbackBuffers can just deal with the COPY_BYTES_PER_ROW requirement internally. Then all the tests can just use ReadbackBuffers without thinking about it.

Reply:
Done.